app/src/main/java/com/sun/moviedb_48/data/source/remote/fetchdata/GetMoviesByCategoriesAsynsTask.java

```diff
@@ -0,0 +1,42 @@
+package com.sun.moviedb_48.data.source.remote.fetchdata;
+
+import android.os.AsyncTask;
+import com.sun.moviedb_48.data.model.Movie;
+
+import com.sun.moviedb_48.data.source.remote.OnFetchDataJsonListener;
+import java.util.ArrayList;
+import org.json.JSONException;
+
+public class GetMoviesByCategoriesAsynsTask extends AsyncTask<String, Void, ArrayList<Movie>> {
+
+    private OnFetchDataJsonListener<Movie> mListener;
+    private Exception mException;
+
+    public GetMoviesByCategoriesAsynsTask(OnFetchDataJsonListener<Movie> listener) {
+        mListener = listener;
+    }
+
+    @Override
+    protected ArrayList<Movie> doInBackground(String... strings) {
+        ArrayList<Movie> movies = null;
+        try {
+            ParseUrlToJson parseUrlToJson = new ParseUrlToJson();
+            String data = parseUrlToJson.getJsonFromUrl(strings[0]);
+            movies = parseUrlToJson.getListMoviesFromJson(data);
+        } catch (Exception e) {
+            e.printStackTrace();
+            mException = e;
+        }
+        return movies;
+    }
+
+    @Override
+    protected void onPostExecute(ArrayList<Movie> movies) {
+        if (mException == null) {
```

Comment:
Đoạn này phải là 
```suggestion
        if (movies == null) {
```
chứ nhỉ?

Reply:
```suggestion
        if (mException != null) {
```

Reply:
Mới đầu em định check cả 2: if(mException == null || movies == null)
Nhưng em lại nghĩ là nếu movies không lấy được data thì nó sẽ nhảy xuống catch rồi quăng exception, lúc đó em gán mException = e
Thì mException sẽ có giá trị, nên chỉ cần xét mException thôi.
...Nhưng cũng không chắc đúng hay không :)))